server/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java

```diff
@@ -1202,6 +1175,16 @@ private static void ensureResultTypesMatch(Collection<? extends Symbol> results)
         }
     }
 
+    @Nullable
+    private static String maybeQuotedSubscript(String columnName) {
```

Comment:
```suggestion
    private static String maybeQuoteSubscript(String columnName) {
```

Reply:
The function will detect a quoted subscript and re-add quoting around the column identifier. So it is not maybe quoting.
The existing name is also confusing and not really correct, new suggestion: `detectAndSanitizeQuotedSubscript`
WDYT?

Reply:
`detectAndSanitizeQuotedSubscript` sounds better, `maybeQuotedSubscript` was a bit confusing